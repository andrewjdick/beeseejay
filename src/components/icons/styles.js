import styled from "styled-components";

export const Svg = styled.svg`
  color: rgba(0, 0, 0, 0.6);
  transition: all 200ms ease-in;

  &:hover {
    color: rgba(0, 0, 0, 0.8);
  }

  &:active {
    transform: scale(0.9);
    color: rgb(0, 0, 0);
  }
`;
