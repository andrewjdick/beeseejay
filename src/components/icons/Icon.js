import React from "react";
import { node, number } from "prop-types";
import { Svg } from "./styles";

export const Icon = ({ width, height, children, ...props }) => (
  <Svg
    id="i-close"
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 32 32"
    width={width}
    height={height}
    fill="none"
    stroke="currentcolor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    {...props}
  >
    {children}
  </Svg>
);

Icon.defaultProps = {
  width: 32,
  height: 32
};

Icon.propTypes = {
  children: node.isRequired,
  width: number,
  height: number
};
