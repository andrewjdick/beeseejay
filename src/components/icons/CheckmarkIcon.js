import React from "react";
import { Icon } from "./Icon";

export const CheckmarkIcon = props => (
  <Icon {...props}>
    <path d="M2 20 L12 28 30 4" />
  </Icon>
);
