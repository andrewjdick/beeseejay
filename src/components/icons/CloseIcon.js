import React from "react";
import { Icon } from "./Icon";

export const CloseIcon = props => (
  <Icon {...props}>
    <path d="M2 30 L30 2 M30 30 L2 2" />
  </Icon>
);
