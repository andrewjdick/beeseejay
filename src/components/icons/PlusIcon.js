import React from "react";
import { Icon } from "./Icon";

export const PlusIcon = props => (
  <Icon {...props}>
    <path d="M16 2 L16 30 M2 16 L30 16" />
  </Icon>
);
