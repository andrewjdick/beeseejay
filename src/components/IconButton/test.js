import React from "react";
import renderer from "react-test-renderer";
import { render } from "react-testing-library";
import "react-testing-library/cleanup-after-each";

import { PlusIcon } from "components/icons/PlusIcon";
import { IconButton } from ".";

// Snapshot testing
it("renders correctly", () => {
  const tree = renderer
    .create(
      <IconButton>
        <PlusIcon />
      </IconButton>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

// Unit testing
it("renders a button element", () => {
  const { container } = render(
    <IconButton>
      <PlusIcon />
    </IconButton>
  );
  expect(container.querySelector("button")).toBeTruthy();
});
