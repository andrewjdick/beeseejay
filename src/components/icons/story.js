import React from "react";
import { storiesOf } from "@storybook/react";
import { number } from "@storybook/addon-knobs";
import { CloseIcon } from "./CloseIcon";
import { PlusIcon } from "./PlusIcon";
import { TrashIcon } from "./TrashIcon";
import { action } from "@storybook/addon-actions";

export const iconSizeOptions = {
  range: true,
  min: 0,
  max: 200,
  step: 1
};

storiesOf("Components/icons", module)
  .add("CloseIcon", () => (
    <CloseIcon
      width={number("width", 150, iconSizeOptions)}
      height={number("height", 150, iconSizeOptions)}
      onClick={action("icon clicked")}
    />
  ))
  .add("PlusIcon", () => (
    <PlusIcon
      width={number("width", 150, iconSizeOptions)}
      height={number("height", 150, iconSizeOptions)}
      onClick={action("icon clicked")}
    />
  ))
  .add("TrashIcon", () => (
    <TrashIcon
      width={number("width", 150, iconSizeOptions)}
      height={number("height", 150, iconSizeOptions)}
      onClick={action("icon clicked")}
    />
  ));
