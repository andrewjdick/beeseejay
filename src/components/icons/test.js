import React from "react";
import renderer from "react-test-renderer";
import { render } from "react-testing-library";
import "react-testing-library/cleanup-after-each";
import "jest-dom/extend-expect";

import { CheckmarkIcon } from "./CheckmarkIcon";
import { Icon } from "./Icon";
import { CloseIcon } from "./CloseIcon";
import { PlusIcon } from "./PlusIcon";
import { TrashIcon } from "./TrashIcon";

// Snapshot Testing
it("renders CheckmarkIcon correctly", () => {
  const tree = renderer.create(<CheckmarkIcon />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders CloseIcon correctly", () => {
  const tree = renderer.create(<CloseIcon />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders PlusIcon correctly", () => {
  const tree = renderer.create(<PlusIcon />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders TrashIcon correctly", () => {
  const tree = renderer.create(<TrashIcon />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Unit Testing
it("Icon defaults to a height and width of 32", () => {
  const { container } = render(
    <Icon>
      <PlusIcon />
    </Icon>
  );

  expect(container.querySelector("svg")).toHaveAttribute("width", "32");
  expect(container.querySelector("svg")).toHaveAttribute("height", "32");
});

it("CheckmarkIcon renders an svg element", () => {
  const { container } = render(<CheckmarkIcon />);
  expect(container.querySelector("svg")).toBeTruthy();
});

it("CloseIcon renders an svg element", () => {
  const { container } = render(<CloseIcon />);
  expect(container.querySelector("svg")).toBeTruthy();
});

it("PlusIcon renders an svg element", () => {
  const { container } = render(<PlusIcon />);
  expect(container.querySelector("svg")).toBeTruthy();
});

it("TrashIcon renders an svg element", () => {
  const { container } = render(<TrashIcon />);
  expect(container.querySelector("svg")).toBeTruthy();
});
