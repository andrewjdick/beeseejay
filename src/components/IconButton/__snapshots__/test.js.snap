// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="sc-bwzfXH jkXeUE"
>
  <svg
    className="sc-bdVaJa iKBuLo"
    fill="none"
    height={32}
    id="i-close"
    stroke="currentcolor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    viewBox="0 0 32 32"
    width={32}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16 2 L16 30 M2 16 L30 16"
    />
  </svg>
</button>
`;
